src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isElementIndex(index)) {
+            return unlink(node(index));
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (object != null) {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (size == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private Node<T> node(int index) {
+        if (index < (size >> 1)) {
+            Node<T> x = firstNode;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = lastNode;
+            for (int i = size - 1; i > index; i--) {
+                x = x.previous;
+            }
+            return x;
+        }
+    }
+
+    private void linkLast(T e) {
+        final Node<T> last = lastNode;
+        final Node<T> newNode = new Node<>(last, e, null);
+        lastNode = newNode;
+        if (last == null) {
+            firstNode = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T e, Node<T> succ) {
+        final Node<T> pred = succ.previous;
+        final Node<T> newNode = new Node<>(pred, e, succ);
+        succ.previous = newNode;
+        if (pred == null) {
+            firstNode = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+
+    private boolean isPositionIndex(int index) {
+        return index >= 0 && index <= size;
+    }
+
+    private boolean isElementIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private T unlink(Node<T> x) {
+        final T element = x.item;
+        final Node<T> next = x.next;
+        final Node<T> prev = x.previous;
+        if (prev == null) {
+            firstNode = next;
+        } else {
+            prev.next = next;
+            x.previous = null;
+        }
+        if (next == null) {
+            lastNode = prev;
+        } else {
+            next.previous = prev;
+            x.next = null;
+        }
+        x.item = null;
+        size--;
+        return element;
     }
```

Comment:
x isn't the best variable name
Also, we don't need to change any of the fields in the node we are unlinking (x in this case, I believe)